crates/bevy_ecs/src/schedule/condition.rs

```diff
@@ -142,3 +161,69 @@ pub mod common_conditions {
         condition.pipe(|In(val): In<bool>| !val)
     }
 }
+
+pub type AndThen<A, B> = CombinatorSystem<AndThenMarker, A, B>;
+
+pub type OrElse<A, B> = CombinatorSystem<OrElseMarker, A, B>;
+
+#[doc(hidden)]
+pub struct AndThenMarker;
+
+impl<In, A, B> Combine<A, B> for AndThenMarker
+where
+    In: Copy,
+    A: System<In = In, Out = bool>,
```

Comment:
Should this be restricted to ReadOnlySystem?

Reply:
Not necessary, `CombinatorSystem<A, B, AndThenMarker>` only implements `ReadOnlySystem` if `A` and `B` do, so the trait bounds "just work" as is.